test/core/codec/leb128_test.cpp

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gtest/gtest.h>
+#include <boost/math/tools/precision.hpp>
+
+#include "codec/leb128/leb128.hpp"
+#include "primitives/boost_multiprecision.hpp"
+
+using namespace boost::multiprecision;
+using namespace boost::math::tools;
+
+/**
+ * @brief Check LEB128 encoding/decoding integrity
+ * @param data value to test
+ */
+template <typename T>
+bool LEB128IntegrityCheck(const T& data)
+{
+  std::vector<uint8_t> encoded = codec::leb128::encode(data);
+  T decoded = codec::leb128::decode<T>(encoded);
+  return data == decoded;
+}
+
+/**
+ * @given LEB128 encode/decode samples of the unsigned values
+ * @when Encoding value with LEB128 codec
+ * @then Produced values must be the same as the given samples
+ */
+TEST(LEB128_Algorithm, sample) {
+  uint32_t decoded{};
+  uint32_t source = 624485;
+  std::vector<uint8_t> sample{0xE5, 0x8E, 0x26};
+  std::vector<uint8_t> encoded = codec::leb128::encode(source);
+  ASSERT_EQ(sample, encoded);  // Check encoding
+  ASSERT_TRUE(true);
+  decoded = codec::leb128::decode<uint32_t>(encoded);
+  ASSERT_EQ(source, decoded);  // Check decoding
+}
+
+
+/**
+ * @given Unsigned values of different variable limits
+ * @When Encoding values with LEB128 codec
+ * @then Decoded values must be the same as source
+ */
+TEST(LEB128_Integrity, uint8_t)
+{
+  ASSERT_TRUE(LEB128IntegrityCheck<uint8_t>(min_value<uint8_t>()));
```

Comment:
You can replace all these tests by single test by using type-parameterized tests feature from gtest